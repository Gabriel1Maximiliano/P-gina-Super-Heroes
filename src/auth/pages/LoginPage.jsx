import React, { useContext } from 'react'
import { useNavigate } from 'react-router-dom';
import { AuhtContext } from '../context/AuhtContext';

export const LoginPage = () => {
  const { login } = useContext( AuhtContext );
  const navigate = useNavigate();

  const handleLogin = () => {
      const lastPath = localStorage.getItem('lastPath') || '/'
    login('Gabriel Loza');

    navigate(lastPath,{
      replace: true
    })
  }
  return (
    <div className='container mt-5'>
        <h1>Login</h1>
        <hr></hr>

        <button className='btn btn-primary'
        
         onClick={handleLogin}>
          Login

        </button>
    </div>
  )
}
