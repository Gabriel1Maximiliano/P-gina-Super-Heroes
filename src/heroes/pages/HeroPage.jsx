import React, { useMemo } from 'react'
import {useParams,Navigate,useNavigate} from 'react-router-dom'
import { getHeroById } from '../helpers/getHeroById';

export const HeroPage = () => {

  const {id} = useParams();

  const navigate = useNavigate()

  const heroe = useMemo(()=>getHeroById(id),[id])// se dispara cada vez q su dependencia cambie  

  
  const handleNavigateBack = () => {
    navigate('/',{
      replace: true
    })
  }
  if(!heroe){
    return <Navigate to='/marvel' />
  }

  return (
    <div className='row mt-5' >
      <div className='col-4' >
        <img
        src={require(`../../imagenes/${id}.jpg`)}
        alt={heroe.superhero}
        className='img-thumbnail'
        />
      </div>
      <div className='col-8' >
        <h3>{ heroe.superhero }</h3>
        <ul className='List-goup list-group-flush' >
          <li>
            <b className='list-group-item' >Alter ego: {heroe.alter_ego}</b>
            <b className='list-group-item' >Publisher: {heroe.publisher}</b>
            <b className='list-group-item' >First Appearence: {heroe.first_appearance}</b>
          </li>
        </ul>
        <h5 className='mt-3' >Characters</h5>
        <p>{ heroe.characters }</p>
        <button className='btn btn-primary' 
        onClick={handleNavigateBack}>
          Back
        </button>
      </div>
    </div>
  )
}
