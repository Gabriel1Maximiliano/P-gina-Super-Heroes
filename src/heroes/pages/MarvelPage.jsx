import React from 'react'
import { HeroList } from '../components/HeroList'


export const MarvelPage = () => {
  const publisher = 'Marvel Comics'
  
  return (
    <>
    <h1>MarvelPage</h1>
    <hr />
    <HeroList publisher={publisher}/>
    </>
  )
}
