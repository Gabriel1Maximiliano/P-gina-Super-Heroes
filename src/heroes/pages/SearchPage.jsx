import React from 'react'
import { useForm } from '../../hooks/useForm'
import { HeroCard } from '../components/HeroCard'
import {useNavigate,useLocation} from 'react-router-dom';
import queryString from 'query-string'
import { getHeroesByName } from '../helpers/getHeroesByName';
export const SearchPage = () => {
const navigate = useNavigate()
const location = useLocation();

const {q=''} = queryString.parse( location.search );
console.log(q)

const heroes = getHeroesByName(q)
    const { searchText, onInputChange,} = useForm({
            searchText:''
        })
        const handleSubmit = (e) => {
            e.preventDefault();
            if(searchText.trim().lenght ===0){
                return
            }
           navigate(`?q=${searchText.toLowerCase().trim()}`)
        }


    return (
        <>
            <h1>SearchPage</h1>

            <h4>Searching</h4>
            <hr />
            <div className='row' >
                <div className='col-5' >
                    <form onSubmit={handleSubmit} >
                        <input type="text"
                            placeholder='Search a heroe'
                            className='form-control'
                            name='searchText'
                            autoComplete='off'
                            value={searchText}
                            onChange={onInputChange}
                        />
                        <button className='btn btn-outline-primary mt-1' 
                        >
                            Search
                        </button>
                    </form>
                </div>

            <div className='col-7' >

                <h4>Results</h4>
                <hr />
                <div className='alert alert-primary' >
                    Search a hero
                </div>
                {
                    heroes.length == 0 && <div className='alert alert-danger'><b>There is no results</b></div>
                }
            
                
            
            {
               heroes && heroes.map(h=>(
                    <HeroCard key={h.id} {...h}/> 
                ))
            }
           {/* <HeroCard />*/}
            </div>
            </div>

        </>
    )
}
