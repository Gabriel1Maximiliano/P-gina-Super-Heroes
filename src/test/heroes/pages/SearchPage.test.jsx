import {screen,useNavigate,fireEvent,render} from '@testing-library/react';
import { scryRenderedComponentsWithType } from 'react-dom/test-utils';

import {MemoryRouter} from 'react-router-dom';
import { SearchPage } from '../../../heroes/pages/SearchPage';


describe('Pruebas sobre <SearchPage />', () => { 
    test('should mostrarse correctamente con valores por defecto ', () => { 
           const {container}=render(
            <MemoryRouter>
              <SearchPage />  
            </MemoryRouter>
           )

          expect( container ).toMatchSnapshot()
     })
     test('should mostrarse correctamente batman con el valor del queryString', () => { 

         render(
             <MemoryRouter initialEntries={['/search?q=batman']} >
           <SearchPage />  
         </MemoryRouter>
        )
        
        expect(screen.getByText('Batman')).toBeTruthy();
        
       
  })
  test('should first', () => { 
    render(
        <MemoryRouter initialEntries={['/search?q=batman']} >
      <SearchPage />  
    </MemoryRouter>
   )
   })
 })