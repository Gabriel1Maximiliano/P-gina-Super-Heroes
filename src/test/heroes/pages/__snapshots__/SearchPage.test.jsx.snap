// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <SearchPage /> should mostrarse correctamente batman con el valor del queryString 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <h4>
    Searching
  </h4>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger"
      >
        <b>
          There is no results
        </b>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas sobre <SearchPage /> should mostrarse correctamente con valores por defecto  1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <h4>
    Searching
  </h4>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger"
      >
        <b>
          There is no results
        </b>
      </div>
    </div>
  </div>
</div>
`;
